include/hffix.hpp

```diff
@@ -604,7 +633,10 @@ class message_writer {
     \param day Day.
     */
     void push_back_date(int tag, int year, int month, int day) {
-        next_ = details::itoa(tag, next_);
+        next_ = details::itoa(tag, next_, buffer_end_);
+        if (buffer_end_ - next_ < details::len("=YYYYMMDD|")) {
```

Comment:
I figured using string literals like this was a robust and self-documenting way of describing the expected length of the output

Reply:
Hah well that's very cute. Maybe a bit too cute. But okay, yeah.